kythe/typescript/indexer.ts

```diff
@@ -753,17 +753,15 @@ class Visitor {
    * getPathFromModule(the './foo' node) might return a string like
    * 'path/to/project/foo'.  See this.moduleName().
    */
-  getModulePathFromModuleReference(sym: ts.Symbol): string {
-    const name = sym.name;
-    // TODO: this is hacky; it may be the case we need to use the TypeScript
-    // module resolver to get the real path (?).  But it appears the symbol
-    // name is the quoted(!) path to the module.
-    // TODO(ayazhafiz): Fix the above.
-    if (!(name.startsWith('"') && name.endsWith('"'))) {
-      this.todo(sym.declarations[0], `TODO: handle module symbol ${name}`);
+  getModulePathFromModuleReference(sym: ts.Symbol): string|undefined {
+    const sf = sym.valueDeclaration;
```

Comment:
Does valueDeclaration always exist/work, even for type-only modules?

Reply:
What is a type only module? .d.ts files?

Reply:
Sorry for being vague.  When I see "valueDeclaration" it makes me think of how a given symbol normally has a 0-1 value declarations and 0-N type declarations, and I was wondering if it's possible here for the value declaration to not exist.  It's weird because this is a module object, but normally isn't it possible for a symbol to not have a value declaration?

Reply:
Apparently not. The types suggest that a symbol always has a value declaration.